calculation_functors.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#ifdef __CUDACC__
+#include <thrust/device_vector.h>
+#include "cuda_types.h"
+#endif
+#include <QVector>
+#include "types.h"
+
+namespace gpusim
+{
+/**
+ * Functor used to perform tanimoto similarity on GPGPU via thrust::transform
+ */
+#ifdef __CUDACC__
+class TanimotoFunctor {
+
+    const int* m_ref_fp;
+    const int m_fp_intsize;
+    const int* m_dbdata;
+
+public:
+
+    TanimotoFunctor(const gpusim::DFingerprint& ref_fp, int fp_intsize, const thrust::device_vector<int>& dbdata) : m_ref_fp(ref_fp.data().get()),m_fp_intsize(fp_intsize),m_dbdata(dbdata.data().get())
+        {};
+
+ /*
+  * This function is defined in calculation_functors.cu as cuda prefers device
+  * code defined in location of use.
+  */
+    __device__ float operator()(const int& fp_index) const;
+};
+#endif
+
+
+/**
+ * Functor used to perform tanimoto similarity on CPU via std::transform
+ */
+struct TanimotoFunctorCPU{
```

Comment:
Same comment here; if you use __host__ __device__ on the operator, don't you only need one class?